pkg/hubble/parser/seven/parser.go

```diff
@@ -176,10 +177,9 @@ func (p *Parser) computeResponseTime(r *accesslog.LogRecord, timestamp time.Time
 func (p *Parser) updateEndpointWorkloads(ip net.IP, endpoint *pb.Endpoint) {
 	if ep, ok := p.endpointGetter.GetEndpointInfo(ip); ok {
 		if pod := ep.GetPod(); pod != nil {
-			olen := len(pod.GetOwnerReferences())
-			endpoint.Workloads = make([]*pb.Workload, olen)
-			for index, owner := range pod.GetOwnerReferences() {
-				endpoint.Workloads[index] = &pb.Workload{Kind: owner.Kind, Name: owner.Name}
+			workload, workloadTypeMeta, ok := utils.GetWorkloadMetaFromPod(pod)
+			if ok {
+				endpoint.Workloads = []*pb.Workload{{Kind: workloadTypeMeta.Kind, Name: workload.Name}}
```

Comment:
Could this be an `append` to the slice?

Reply:
This is the only place it's used besides the other parser, so is that more style or for ensuring if something else ends up setting this field, we're not overwriting it?

Reply:
More for the latter, but if it's not being used elsewhere, then maybe `append` is not necessary. `append` is just an approach which doesn't assume the state of the field.

Reply:
I think I'll keep it how it is now then. If we end up needing to source this from multiple places, then I suspect just appending won't necessarily be correct either, since we may end up with duplicates, etc.